### PR TITLE
Upgrade of pylint to 2.9.3 and python 3.9

### PR DESCRIPTION
**Pull request by jgmills**
Pylint is a Python static code analysis tool.  This PR upgrades pylint from versions 0.25.2 to 2.9.3 and the python version it uses from 2.7to 3.9 .  It also obsoletes the old pylint package.  This is a major upgrade of pylint: the old one is nine years old.

The changes to Makefile accomodate the software version changes, the new build style, and the results of built-in tests.  There are no patches, as they are not necessary for this version.  There are also no man pages, as they are not supplied with this version.  A user manual is available at:

    https://pylint.pycqa.org/

The manifest file pylint-39.p5m contains new explicit dependancies of pylint.  The manifest file pylint.p5m includes only documentation files.  The manifests/sample-manifest.p5m file was also updated for the new software version.

This PR leaves two dependancies unused: library/python/logilab-common-27 and library/python/logilab-astng-27 .  They will be obsoleted in a separate PR.

The build, install, and publish steps were all successful.  Running "gmake test" produced the output included in Makefile, with 14 failed tests and 1364 passed tests.  I also tried an external test of the new pylint, using the example file included in the pylint tutorial.  It produced the same messages as in the tutorial, showing that pylint actually works.

### CODE REVIEW

File: components/python/pylint/Makefile [hunk 1]
```diff
@@ -78,47 +82,35 @@ COMPONENT_POST_INSTALL_ACTION += \
...
+#=========================== short test summary info ============================
+#FAILED test_import_graph.py::test_dependencies_graph[tests/regrtest_data/foo.dot]
+#FAILED test_self.py::TestRunTC::test_pkginfo - AssertionError: expected outpu...
+#FAILED test_self.py::TestRunTC::test_enable_all_works - assert "*************...
+#...
+#ERROR benchmark/test_baseline_benchmarks.py::TestEstablishBaselineBenchmarks::test_baseline_benchmark_j10_single_working_checker
+#ERROR benchmark/test_baseline_benchmarks.py::TestEstablishBaselineBenchmarks::test_baseline_benchmark_j1_all_checks_single_file
+#ERROR benchmark/test_baseline_benchmarks.py::TestEstablishBaselineBenchmarks::test_baseline_benchmark_j1_all_checks_lots_of_files
+#= 14 failed, 1364 passed, 59 skipped, 200 deselected, 11 errors in 185.41s (0:03:05) =
+
```
Comment (AndWac) [inline components/python/pylint/Makefile:125]: You should also add the test dependencies, eg.:
\# Manually added test dependencies
REQUIRED_PACKAGES += library/python/flaky
REQUIRED_PACKAGES += library/python/hypothesis
REQUIRED_PACKAGES += library/python/isort
REQUIRED_PACKAGES += library/python/mccabe
REQUIRED_PACKAGES += library/python/pluggy
REQUIRED_PACKAGES += library/python/py
REQUIRED_PACKAGES += library/python/pytest
REQUIRED_PACKAGES += library/python/pytest-benchmark
REQUIRED_PACKAGES += library/python/pytest-reporter

Comment (jgmills) [reply components/python/pylint/Makefile:125]: Now, I don't know what to do.  I've never seen much benefit to listing test dependancies that way.

Only four of the packages listed above (library/python/isort library/python/mccabe library/python/py library/python/pytest) are installed on my development system.  I certainly could add them.

The other five (library/python/flaky library/python/hypothesis library/python/pluggy library/python/pytest-benchmark library/python/pytest-reporter) are not installed.  I assume they are not necessary, although installing might change the test result.  I'd prefer not to add them.

Comment (AndWac) [reply components/python/pylint/Makefile:125]: The purpose of this is to have the tests reproduceable. It's the intersection of what pyling uses and we have. If you don't state the test requirements every other developer will have other test results.
Even you can have different test results if you run them on a different machine.

Comment (jgmills) [reply components/python/pylint/Makefile:125]: Okay, I've added the test dependancies.  The only change was in the test results.  The errors count, formerly 11, has disappeared.  The passed count increased by 11.  The build, install, and publish steps were unchanged.

**Pull request merged**